swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -230,15 +290,16 @@ extension WorkflowNode.SubtreeManager {
 
         private var outputMap: (W.Output) -> AnyWorkflowAction<WorkflowType>
 
-        private var disposable: Disposable? = nil
+        private let (lifetime, token) = Lifetime.make()
 
         init(worker: W, outputMap: @escaping (W.Output) -> AnyWorkflowAction<WorkflowType>) {
             self.worker = worker
             self.signalProducer = worker.run()
             self.outputMap = outputMap
             super.init()
 
-            disposable = signalProducer
+            signalProducer
+                .take(during: lifetime)
```

Comment:
This change has a worker get cancelled if it's not still declared on a subsequent render pass. See the test in `SubtreeManagerTests` for it.

(Alternatively, this could have been done by holding onto the `disposable`, and calling `disposable?.dispose()` in `deinit()`. I'm open to either approach if someone has strong feelings)

Reply:
In Rx, we tend to use the latter pattern if we need to re-subscribe later. We add disposables to a `CompositeDisposable` and clear it if it's a one-shot thing.

Reply:
I think this feels appropriate for ReactiveSwift, and the lifetimes (if it was RxSwift, there would be a "disposeBag" (neat name) that would also get rid of these as the disposable when the object gets deallocated).  Since we are never unsubscribing from these except with the object goes away, I'm not going to add the extra code to get rid of it in `deinit`